src/config/ConfigService.js

```diff
@@ -95,41 +105,37 @@ class Config {
 	constructor(app) {
 		this._dynamic_config = {};
 		this._defaults = {};
+		this._listeners = [];
 
 		if (typeof LOCAL_CONFIG !== 'undefined') {
 			this._local_config = LOCAL_CONFIG;
 		} else {
 			this._local_config = {};
 		}
-
 	}
 
-
 	get(key) {
 		// First check the remote config
-		if (this._dynamic_config[key] != undefined) {
+		if (this._dynamic_config[key] !== undefined) {
 			return this._dynamic_config[key];
-		};
-
+		}
 		// Then check the local config
-		if (this._local_config[key] != undefined) {
-			this._local_config[key];
-		};
-
+		if (this._local_config[key] !== undefined) {
+			return this._local_config[key];
+		}
 		// And finally, check defaults
-		if (this._defaults[key] != undefined) {
+		if (this._defaults[key] !== undefined) {
 			return this._defaults[key];
-		};
-
+		}
 		return undefined;
 	}
 
+	getMergedConfig() {
+		return Object.assign({}, this._defaults, this._local_config, this._dynamic_config);
+	}
 
-	dispatch(store) {
-		var config = Object.assign({}, this._defaults, this._local_config, this._dynamic_config);
-		store.dispatch({
-			type: CHANGE_CONFIG,
-			config: config
-		});
+	_notifyChange() {
+		const merged = this.getMergedConfig();
+		this._listeners.forEach(fn => fn(merged));
```

Comment:
Remove listeneres, try to use the `dispatch` from the store directly even within the class (like `App.Store.dispatch(...)`, or find a way via any top level application syncer